datafusion/expr/src/expr.rs

```diff
@@ -2607,11 +2793,23 @@ pub(crate) fn schema_name_from_exprs_comma_separated_without_space(
     schema_name_from_exprs_inner(exprs, ",")
 }
 
+/// Get `sql_name` for Vector of expressions.
```

Comment:
Instead of adding so many new so specific functions, what do you think about adding something like the following. I think the API would be easier to understand and it would also let us save a `String` copy when formatting as sql

```rust
/// Formats a list of `&Expr` with a custom separator
struct ExprListDisplay<'a> {
  exprs: &'a [Expr],
  sep: &'a str,
}
...
impl <'a> Display for  ExprListDisplay<'a> {
...
}
```